test/Microsoft.AspNetCore.Mvc.DataAnnotations.Test/Internal/DataAnnotationsMetadataProviderTest.cs

```diff
@@ -727,6 +782,118 @@ public void CreateDisplayMetadata_IsFlagsEnum_ReflectsModelType(Type type, bool
                 .ThenBy(item => item.Key.Name, StringComparer.Ordinal));
         }
 
+        private DataAnnotationsMetadataProvider CreateLocalizingProvider()
+        {
+            var stringLocalizer = new Mock<IStringLocalizer>(MockBehavior.Strict);
+            stringLocalizer
+                .Setup(loc => loc[It.IsAny<string>()])
+                .Returns<string>((k =>
+                {
+                    if (k.Contains("Loc_Two"))
+                    {
+                        return new LocalizedString(k, $"{k} {CultureInfo.CurrentCulture}");
+                    }
+                    else
+                    {
+                        return new LocalizedString(k, k, resourceNotFound: true);
+                    }
+                }));
+
+            var stringLocalizerFactory = new Mock<IStringLocalizerFactory>(MockBehavior.Strict);
+            stringLocalizerFactory
+                .Setup(factory => factory.Create(typeof(EnumWithLocalizedDisplayNames)))
+                .Returns(stringLocalizer.Object);
+
+            return new DataAnnotationsMetadataProvider(stringLocalizerFactory: stringLocalizerFactory.Object);
+        }
+
+        [Fact]
+        public void CreateDisplayMetadata_EnumGroupedDisplayNamesAndValues_IStringLocalizer()
+        {
+            // Arrange
+            var provider = CreateLocalizingProvider();
+
+            var key = ModelMetadataIdentity.ForType(typeof(EnumWithLocalizedDisplayNames));
+            var attributes = new object[0];
+
+            // Act
+            var context = new DisplayMetadataProviderContext(key, new ModelAttributes(attributes));
+            provider.CreateDisplayMetadata(context);
+
+            string frenchEnumDisplay;
+            using (new CultureReplacer("fr-FR", "fr-FR"))
+            {
+                frenchEnumDisplay = context.DisplayMetadata.EnumGroupedDisplayNamesAndValues
+                    .Where(kvp => kvp.Value == "2")
+                    .First().Key.Name;
```

Comment:
If the localizer always localized, could instead save `EnumGroupedDisplayNamesAndValues.Select(kvp => kvp.Key.Name).ToList()`. Then `Assert.All(frenchEnumNames, name => Assert.EndsWith(" fr-FR", name, StringComparison.Ordinal)` down below.

For future reference, don't separate out the `Where()` condition; use `First(kvp => kvp.Value == "2")`.